filters/openpolicyagent/internal/envoy/skipperadapter.go

```diff
@@ -35,9 +39,29 @@ func AdaptToExtAuthRequest(req *http.Request, metadata *ext_authz_v3_core.Metada
 		},
 	}
 
-	if !utf8.ValidString(ereq.Attributes.Request.Http.Path) {
-		return nil, fmt.Errorf("invalid utf8 in path: %q", ereq.Attributes.Request.Http.Path)
+	return ereq, nil
+}
+
+func buildPathWithQueryParams(req *http.Request) string {
+	if req.URL.RawQuery == "" {
+		return url.PathEscape(req.URL.Path)
 	}
+	return url.PathEscape(req.URL.Path) + "?" + req.URL.RawQuery
```

Comment:
do we need to use `url.QueryEscape`?

Reply:
[req.URL.RawQuery](https://cs.opensource.google/go/go/+/refs/tags/go1.23.1:src/net/url/url.go;l=379) is already encoded, so didn't have to encode it again.

Reply:
Are you sure you want to have this result: https://go.dev/play/p/AOiC7-DQEBs ?

Reply:
> Are you sure you want to have this result: https://go.dev/play/p/AOiC7-DQEBs ? path and the query (query does not look encoded to me, but maybe at http.Server it is encoded)

To be honest, I am not entirely sure about how Go handles the parsing when there is a UTF invalid character passed in the URL string. (IINM, to be a valid URL, non-ascii characters needs to be percentage encoded)

Prior to processing the URL, i think in skipper we check for UTF invalid characters. The [check](https://github.com/zalando/skipper/pull/3207/files#diff-dee220a324e25f9867826168abe29e559c7291d044fe87d44165c3ef28384c5cR15) was added to do the same to return a 400 for a URLs with invalid UTF characters.

To test the behaviour of query string, I tried an example, where there is a space in the query param value
https://go.dev/play/p/Z1qeKvyfMre

This shows that the req.URL.RawQuery returns the unescaped query string.

Reply:
And the PathEscape() is by intention, right?
So `/` is `%2F`

Reply:
Yes, it was intentional.

When OPA creates the input to policy it unescapes the path and query. So I didn't see a problem with this. 
The reason to PathEscape() was to avoid a double escape/decode.